lib/court_data_adaptor/resource/base.rb

```diff
@@ -9,7 +9,13 @@ def self.call(env)
 
   module Resource
     class Base < JsonApiClient::Resource
+      include JsonApiClient::Helpers::Callbacks
       include Configurable
+      include ActsAsResource
+
+      before_save do
+        refresh_token_if_required!
+      end
```

Comment:
i thought this did break tests. checking..

Reply:
I added a commit and the [test suite definitely fails](https://app.circleci.com/pipelines/github/ministryofjustice/laa-court-data-ui/833/workflows/a65b6221-50b6-46b9-8a9f-e80ee761c7e0/jobs/2729/parallel-runs/0/steps/0-113) if you remove `refresh_token_if_required!` from the callback

This [shared spec](https://github.com/ministryofjustice/laa-court-data-ui/pull/174/files#diff-b70817e86ea56d7269f21388f0314fe7R19-R34) covers that

However feature specs and request specs don't fail because of stubbing. Thinking to add one.